instrumentation/netty/netty-3.8/src/main/java/io/opentelemetry/auto/instrumentation/netty/v3_8/server/HttpServerRequestTracingHandler.java

```diff
@@ -48,43 +45,25 @@ public void messageReceived(final ChannelHandlerContext ctx, final MessageEvent
         contextStore.putIfAbsent(ctx.getChannel(), ChannelTraceContext.Factory.INSTANCE);
 
     if (!(msg.getMessage() instanceof HttpRequest)) {
-      final Span span = channelTraceContext.getServerSpan();
-      if (span == null) {
-        ctx.sendUpstream(msg); // superclass does not throw
+      Context serverSpanContext = TRACER.getServerSpanContext(channelTraceContext);
+      if (serverSpanContext == null) {
+        ctx.sendUpstream(msg);
       } else {
-        try (final Scope scope = TRACER.withSpan(span)) {
-          ctx.sendUpstream(msg); // superclass does not throw
+        try (final Scope ignored = ContextUtils.withScopedContext(serverSpanContext)) {
+          ctx.sendUpstream(msg);
         }
       }
       return;
     }
 
     final HttpRequest request = (HttpRequest) msg.getMessage();
 
-    final Span.Builder spanBuilder =
-        TRACER.spanBuilder(DECORATE.spanNameForRequest(request)).setSpanKind(SERVER);
-    final SpanContext extractedContext = extract(request.headers(), GETTER);
-    if (extractedContext.isValid()) {
-      spanBuilder.setParent(extractedContext);
-    } else {
-      // explicitly setting "no parent" in case a span was propagated to this thread
-      // by the java-concurrent instrumentation when the thread was started
-      spanBuilder.setNoParent();
-    }
-    final Span span = spanBuilder.startSpan();
-    try (final Scope scope = TRACER.withSpan(span)) {
-      DECORATE.afterStart(span);
-      DECORATE.onConnection(span, ctx.getChannel());
-      DECORATE.onRequest(span, request);
-
-      channelTraceContext.setServerSpan(span);
-
+    Span span = TRACER.startSpan(request, ctx.getChannel(), "netty.request", null);
+    try (final Scope ignored = TRACER.startScope(span, channelTraceContext)) {
```

Comment:
can merge inner try/catch into the outer try-with-resources (and i think same in v4_0 and v4_1 copies)